File: Max1DRangeSum.java
import java.util.Scanner;

public class Max1DRangeSum{
	public static void main(String[] args) {
	    int sum = 0, ans = 0;
	    String line;

	    Scanner scan = new Scanner(System.in);


	    System.out.println("Aguarde... O sistema está encontrando o valor da maior sequência.");
	    while(scan.hasNext()){
	    	line = scan.next();
	        if (line.equals("end"))
	            break;

	    	int number = Integer.parseInt(line);

	        sum += number;           
	        ans = max(ans, sum);            
	        if(sum < 0)  sum = 0;           
	    }                                  

	    System.out.println("A soma eh: " + ans);
	    System.out.println("\nObrigado por utilizar!");

	}

	public static int max(int x, int y){
		return (x < y) ? y : x;
	}
}

File: max1DRangeSum.cpp
#include <iostream>
#include <string>
#include <vector>
#include <sstream>

using namespace std;

int  main(){
    int sum = 0, ans = 0;
    string line;


    cout << "Aguarde... O sistema está encontrando o valor da maior sequência." <<endl;
    while(cin >> line){

        if (line == "end")
            break;

    	int number;
        istringstream ( line ) >> number;

        sum += number;           
        ans = max(ans, sum);            
        if(sum < 0)  sum = 0;           
    }                                  

    cout << "A soma eh: " << ans << endl;
    cout << "\nObrigado por utilizar!" << endl;

    return 0;
}